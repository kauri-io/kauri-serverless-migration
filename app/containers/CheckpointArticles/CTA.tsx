import { Component } from 'react'
import Loading from '../../components/Loading'
import TertiaryButton from '../../components/Button/TertiaryButton'
export const CheckpointArticlesIcon = () => (
    <svg
        width="20"
        height="20"
        viewBox="0 0 20 20"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
    >
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20Z"
            fill="#3897F0"
        />
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M13.7647 11.0714L9.88235 4L6 11.0714L9.88235 16L13.7647 11.0714ZM9.8823 6.24008L12.3689 10.7686L9.8823 11.8947L7.39571 10.7686L9.8823 6.24008ZM11.3343 12.4182L9.88242 13.0751L8.43051 12.4177L9.88242 14.2607L11.3343 12.4182Z"
            fill="white"
        />
    </svg>
)

interface IState {
    activated: boolean
}

interface IProps {
    pageType: string
    checkpointArticlesAction: () => void
}

class CTA extends Component<IProps, IState> {
    constructor(props) {
        super(props)
        this.state = {
            activated: false,
        }
    }
    componentDidMount() {
        this.loadWeb3()
    }

    async loadWeb3() {
        const sm = await import('../../lib/init-smart-contracts')
        sm.initSmartContracts(global.window.web3, (result, err) => {
            if (err) {
                console.error(err)
                this.setState({ activated: true })
            }
            if (result) {
                this.setState({ activated: true })
            }
        })
    }

    render() {
        if (this.state.activated === false) {
            return <Loading />
        }
        return (
            <TertiaryButton
                color={'textPrimary'}
                icon={<CheckpointArticlesIcon />}
                onClick={() => this.props.checkpointArticlesAction()}
            >
                {this.props.pageType === 'public-profile'
                    ? 'Submit Articles On-chain'
                    : 'Submit Article On-chain'}
            </TertiaryButton>
        )
    }
}

export default CTA
