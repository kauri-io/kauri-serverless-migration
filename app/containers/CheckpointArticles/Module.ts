import { showNotificationAction } from '../../lib/Epics/ShowNotificationEpic'
import { checkpointArticles } from '../../queries/Article'
import analytics from '../../lib/analytics'
import { IDependencies, IReduxState } from '../../lib/Module'
import { of, from } from 'rxjs'
import { ofType, Epic } from 'redux-observable'
import { switchMap, mergeMap, tap, mapTo, catchError } from 'rxjs/operators'

const CHECKPOINT_ARTICLES = 'CHECKPOINT_ARTICLES'

export interface CheckpointArticlesAction {
    type: string
}

type CheckpointArticlesCommandOutput = {
    merkleRoot: string
    checkpointHash: string
    signatureV: string
    signatureR: string
    signatureS: string
}

type CheckpointArticlesCommandResponse = {
    data: { output: CheckpointArticlesCommandOutput }
}

interface ICheckpointArticlesCommandOutput {
    hash: string
}

export const checkpointArticlesAction: () => CheckpointArticlesAction = () => ({
    type: CHECKPOINT_ARTICLES,
})

export const checkpointArticlesEpic: Epic<
    CheckpointArticlesAction,
    any,
    IReduxState,
    IDependencies
> = (
    action$,
    _,
    {
        apolloClient,
        smartContracts,
        web3,
        web3GetNetwork,
        apolloSubscriber,
        getGasPrice,
    }
) =>
    action$.pipe(
        ofType(CHECKPOINT_ARTICLES),
        switchMap(() =>
            web3GetNetwork()
                .mergeMap(() =>
                    apolloClient.mutate({
                        mutation: checkpointArticles,
                        variables: {},
                    })
                )
                .flatMap(({ data: { checkpointArticles: { hash } } }) =>
                    from(apolloSubscriber(hash))
                )
                .do(h => console.log(h))
                .switchMap(
                    ({
                        data: {
                            output: {
                                merkleRoot,
                                checkpointHash,
                                signatureV,
                                signatureR,
                                signatureS,
                            },
                        },
                    }) =>
                        from(getGasPrice()).pipe(
                            mergeMap<number, string>(gasPrice =>
                                (smartContracts().KauriCore
                                    .checkpointArticles as any).sendTransaction(
                                    merkleRoot,
                                    checkpointHash,
                                    signatureV,
                                    signatureR,
                                    signatureS,
                                    {
                                        from: web3.eth.accounts[0],
                                        value: 0,
                                        gas: 250000,
                                        gasPrice,
                                    }
                                )
                            ),
                            mergeMap(transactionHash =>
                                apolloSubscriber<
                                    ICheckpointArticlesCommandOutput
                                >(transactionHash, 'ArticlesCheckpointed')
                            ),
                            tap(() => apolloClient.resetStore()),
                            mapTo(
                                showNotificationAction({
                                    notificationType: 'success',
                                    message: 'Articles checkpointed!',
                                    description:
                                        'All Kauri platform articles are now On-chain!',
                                })
                            ),
                            tap(() => {
                                analytics.track('Checkpoint', {
                                    category: 'generic',
                                })
                            }),
                            catchError(err => {
                                if (
                                    err.message &&
                                    err.message.includes('locked')
                                ) {
                                    return of(
                                        showNotificationAction({
                                            notificationType: 'error',
                                            message: 'Your wallet is locked!',
                                            description:
                                                'Please unlock your wallet!',
                                        })
                                    )
                                } else if (
                                    (err.message &&
                                        err.message.includes(
                                            "'KauriCore' of undefined"
                                        )) ||
                                    err.message.includes('Wrong network')
                                ) {
                                    return of(
                                        showNotificationAction({
                                            notificationType: 'error',
                                            message: 'Wrong network',
                                            description:
                                                'Please switch to the Rinkeby network to checkpoint articles!',
                                        })
                                    )
                                } else if (
                                    err.message &&
                                    err.message.includes(
                                        'Wrong metamask account'
                                    )
                                ) {
                                    return of(
                                        showNotificationAction({
                                            notificationType: 'error',
                                            message: 'Wrong metamask account',
                                            description:
                                                'Please switch to the correct account!',
                                        })
                                    )
                                }
                                console.error(err)
                                return of(
                                    showNotificationAction({
                                        notificationType: 'error',
                                        message: 'Submission error',
                                        description: 'Please try again!',
                                    })
                                )
                            })
                        )
                )
                .catch(err => {
                    if (err.message && err.message.includes('locked')) {
                        return of(
                            showNotificationAction({
                                notificationType: 'error',
                                message: 'Your wallet is locked!',
                                description: 'Please unlock your wallet!',
                            })
                        )
                    } else if (
                        (err.message &&
                            err.message.includes("'KauriCore' of undefined")) ||
                        err.message.includes('Wrong network')
                    ) {
                        return of(
                            showNotificationAction({
                                notificationType: 'error',
                                message: 'Wrong network',
                                description:
                                    'Please switch to the Rinkeby network to checkpoint articles!',
                            })
                        )
                    } else if (
                        err.message &&
                        err.message.includes('Wrong metamask account')
                    ) {
                        return of(
                            showNotificationAction({
                                notificationType: 'error',
                                message: 'Wrong metamask account',
                                description:
                                    'Please switch to the correct account!',
                            })
                        )
                    }
                    console.error(err)
                    return of(
                        showNotificationAction({
                            notificationType: 'error',
                            message: 'Submission error',
                            description: 'Please try again!',
                        })
                    )
                })
        )
    )
