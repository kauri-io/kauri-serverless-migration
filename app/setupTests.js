import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
