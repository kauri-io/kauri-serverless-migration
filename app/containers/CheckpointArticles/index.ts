import View from './View'
import { connect } from 'react-redux'
import { checkpointArticlesAction } from './Module'
import { IReduxState } from '../../lib/Module'

const mapStateToProps = (state: IReduxState) => ({
    hostName: state.app && state.app.hostName,
})

export default connect(mapStateToProps, { checkpointArticlesAction })(View)
