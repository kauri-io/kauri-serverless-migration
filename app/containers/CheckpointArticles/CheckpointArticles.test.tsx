import CheckpointArticles, { IProps } from './View'
import { mountWithRedux } from '../../setupTests'
import { MockedProvider } from 'react-apollo/test-utils'

describe('containers/CheckpointArticles', () => {
    it('should match snapshot', () => {
        const props = {
            checkpointArticlesAction: jest.fn(),
            pageType: 'public-profile' as IProps['pageType'],
            isOwner: true,
        }
        const wrapper = mountWithRedux(
            <MockedProvider>
                <CheckpointArticles {...props} />
            </MockedProvider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
