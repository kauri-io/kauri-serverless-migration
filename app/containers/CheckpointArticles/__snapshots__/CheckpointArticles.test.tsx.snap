// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CheckpointArticles should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {
          "MuiButton": Object {
            "contained": Object {
              "&&:disabled": Object {
                "backgroundColor": "#e0e0e0",
              },
            },
            "outlined": Object {
              "&&:disabled": Object {
                "backgroundColor": "#e0e0e0",
              },
            },
          },
        },
        "palette": Object {
          "action": Object {
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "hover": "rgba(0, 0, 0, 0.08)",
            "hoverOpacity": 0.08,
            "selected": "rgba(0, 0, 0, 0.14)",
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#f5f5f5",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#1E2428",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "primary": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "rgb(7, 118, 93)",
            "light": "rgb(59, 186, 158)",
            "main": "#0BA986",
          },
          "secondary": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "rgb(178, 178, 178)",
            "light": "rgb(255, 255, 255)",
            "main": "#ffffff",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
          "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
          "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
          "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
          "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.5,
          },
          "body2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "letterSpacing": "0.01071em",
            "lineHeight": 1.43,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.02857em",
            "lineHeight": 1.75,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.03333em",
            "lineHeight": 1.66,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightBold": 700,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.75rem",
            "fontWeight": 300,
            "letterSpacing": "-0.00833em",
            "lineHeight": 1,
          },
          "h3": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.04,
          },
          "h4": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "letterSpacing": "0.00735em",
            "lineHeight": 1.17,
          },
          "h5": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.33,
          },
          "h6": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.25rem",
            "fontWeight": 500,
            "letterSpacing": "0.0075em",
            "lineHeight": 1.6,
          },
          "htmlFontSize": 16,
          "overline": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.08333em",
            "lineHeight": 2.66,
            "textTransform": "uppercase",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subtitle1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.75,
          },
          "subtitle2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.00714em",
            "lineHeight": 1.57,
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "speedDial": 1050,
          "tooltip": 1500,
        },
        Symbol(mui.nested): false,
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "DEFAULT_CARD_WIDTH": 288,
          "MAIN_BODY_WIDTH": 1272,
          "bg": Object {
            "bgPrimary": "#1E2428",
            "bgSecondary": "#1E3D3B",
            "borderTextArea": "#979797",
            "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
            "disabledBackgroundColor": "#D6D6D6",
            "disabledTextColor": "#9B9B9B",
            "divider": "#EBEBEB",
            "errorRedColor": "#C03030",
            "hoverTextColor": "#108E72",
            "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
            "paddingTop": "2.5em",
            "primary": "#0BA986",
            "primaryColor": "#0BA986",
            "primaryDark": "#11856B",
            "primaryTextColor": "#1E2428",
            "secondary": "#11856B",
            "secondaryBlue": "rgb(81, 123, 162, 1)",
            "secondaryBlueDark": "rgb(81, 118, 142, 1)",
            "secondaryColor": "#1E3D3B",
            "secondaryTextColor": "#283035",
            "tertiaryBackgroundColor": "#F7F7F7",
            "textPrimary": "#1E2428",
            "white": "#FFFFFF",
          },
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "breakpoints": Array [
            "500px",
            "52em",
            "1272px",
          ],
          "colors": Object {
            "bgPrimary": "#1E2428",
            "bgSecondary": "#1E3D3B",
            "borderTextArea": "#979797",
            "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
            "disabledBackgroundColor": "#D6D6D6",
            "disabledTextColor": "#9B9B9B",
            "divider": "#EBEBEB",
            "errorRedColor": "#C03030",
            "hoverTextColor": "#108E72",
            "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
            "paddingTop": "2.5em",
            "primary": "#0BA986",
            "primaryColor": "#0BA986",
            "primaryDark": "#11856B",
            "primaryTextColor": "#1E2428",
            "secondary": "#11856B",
            "secondaryBlue": "rgb(81, 123, 162, 1)",
            "secondaryBlueDark": "rgb(81, 118, 142, 1)",
            "secondaryColor": "#1E3D3B",
            "secondaryTextColor": "#283035",
            "tertiaryBackgroundColor": "#F7F7F7",
            "textPrimary": "#1E2428",
            "white": "#FFFFFF",
          },
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "fontSizes": Array [
            11,
            13,
            14,
            16,
            18,
            20,
            22,
            28,
            32,
          ],
          "fontWeight": Array [
            300,
            "normal",
            500,
            700,
          ],
          "hoverTextColor": "#108E72",
          "padContent": "
    max-width: 1272px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1272px) {
      min-width: 1272px;
    }
    @media (max-width: 1272px) {
      min-width: 90%;
    }
  ",
          "padding": "calc((100vw - 1272px) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "space": Array [
            0,
            10,
            15,
            30,
            60,
          ],
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        }
      }
    >
      <MockedProvider
        addTypename={true}
      >
        <ApolloProvider
          client={
            ApolloClient {
              "__operations_cache__": Map {},
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "clearStoreCallbacks": Array [],
              "clientAwareness": Object {},
              "defaultOptions": Object {},
              "disableNetworkFetches": false,
              "link": ApolloLink {
                "request": [Function],
              },
              "mutate": [Function],
              "query": [Function],
              "queryDeduplication": true,
              "reFetchObservableQueries": [Function],
              "resetStore": [Function],
              "resetStoreCallbacks": Array [],
              "ssrMode": false,
              "store": DataStore {
                "cache": InMemoryCache {
                  "addTypename": true,
                  "cacheKeyRoot": KeyTrie {
                    "weakness": true,
                  },
                  "config": Object {
                    "addTypename": true,
                    "dataIdFromObject": [Function],
                    "fragmentMatcher": HeuristicFragmentMatcher {},
                    "freezeResults": false,
                    "resultCaching": true,
                  },
                  "data": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "maybeBroadcastWatch": [Function],
                  "optimisticData": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "silenceBroadcast": false,
                  "storeReader": StoreReader {
                    "executeSelectionSet": [Function],
                    "executeStoreQuery": [Function],
                    "executeSubSelectedArray": [Function],
                    "freezeResults": false,
                  },
                  "storeWriter": StoreWriter {},
                  "typenameDocumentCache": Map {},
                  "watches": Set {},
                },
              },
              "version": "2.4.13",
              "watchQuery": [Function],
            }
          }
        >
          <CheckpointArticles
            checkpointArticlesAction={[MockFunction]}
            isOwner={true}
            pageType="public-profile"
          >
            <View__CheckpointedArticlesContainer>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "View__CheckpointedArticlesContainer-sc-1eullvd-0",
                      "isStatic": false,
                      "lastClassName": "dIJWGf",
                      "rules": Array [
                        "display:flex;>:first-child{margin-right:",
                        [Function],
                        "px;color:",
                        [Function],
                        ";}font-size:",
                        [Function],
                        "px;font-weight:",
                        [Function],
                        ";text-transform:uppercase;align-items:center;justify-content:center;width:100%;",
                      ],
                    },
                    "displayName": "View__CheckpointedArticlesContainer",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "View__CheckpointedArticlesContainer-sc-1eullvd-0",
                    "target": "div",
                    "toString": [Function],
                    "usesTheme": true,
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="View__CheckpointedArticlesContainer-sc-1eullvd-0 dIJWGf"
                >
                  <AllArticlesOnMainnet>
                    <View__CheckpointedArticlesContainer>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "View__CheckpointedArticlesContainer-sc-1eullvd-0",
                              "isStatic": false,
                              "lastClassName": "dIJWGf",
                              "rules": Array [
                                "display:flex;>:first-child{margin-right:",
                                [Function],
                                "px;color:",
                                [Function],
                                ";}font-size:",
                                [Function],
                                "px;font-weight:",
                                [Function],
                                ";text-transform:uppercase;align-items:center;justify-content:center;width:100%;",
                              ],
                            },
                            "displayName": "View__CheckpointedArticlesContainer",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "View__CheckpointedArticlesContainer-sc-1eullvd-0",
                            "target": "div",
                            "toString": [Function],
                            "usesTheme": true,
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="View__CheckpointedArticlesContainer-sc-1eullvd-0 dIJWGf"
                        >
                          <CheckpointArticlesIcon>
                            <svg
                              fill="none"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clipRule="evenodd"
                                d="M10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20Z"
                                fill="#3897F0"
                                fillRule="evenodd"
                              />
                              <path
                                clipRule="evenodd"
                                d="M13.7647 11.0714L9.88235 4L6 11.0714L9.88235 16L13.7647 11.0714ZM9.8823 6.24008L12.3689 10.7686L9.8823 11.8947L7.39571 10.7686L9.8823 6.24008ZM11.3343 12.4182L9.88242 13.0751L8.43051 12.4177L9.88242 14.2607L11.3343 12.4182Z"
                                fill="white"
                                fillRule="evenodd"
                              />
                            </svg>
                          </CheckpointArticlesIcon>
                          <span>
                            All Articles On-chain
                          </span>
                        </div>
                      </StyledComponent>
                    </View__CheckpointedArticlesContainer>
                  </AllArticlesOnMainnet>
                </div>
              </StyledComponent>
            </View__CheckpointedArticlesContainer>
          </CheckpointArticles>
        </ApolloProvider>
      </MockedProvider>
    </ThemeProvider>
  </ThemeProvider>
</Provider>
`;
